Do not show offset parameter in the error message when there is no offset

Consider the following code:
`class errmsg{
    public void abc(String a)
    {
        int x = a.length();
        char z;
        for(int i = 0; i < x; i++)
            z = a.charAt(i + 1);
    }
}`

Run it by typing in the terminal:
javac -processor org.checkerframework.checker.index.IndexChecker errmsg.java


Output:
errmsg.java:7: error: [argument.type.incompatible] incompatible types in argument.
            z = a.charAt(i + 1);
                           ^
  found   : @LTEqLengthOf("a") int
  required: @LTLengthOf(value="a", offset={}) int
1 error

If there is no offset in LTEqLengthOf() or LTLengthOf(), the offset parameter is redundant. So, why does the error message show: required: offset={}, a null set of offset.
So, I suggest the offset parameter to not be shown in both the found and the required expressions when there is no offset for consistency, organizations and less distraction from the actual matter.

--- a/framework/src/main/java/org/checkerframework/framework/util/DefaultAnnotationFormatter.java
+++ b/framework/src/main/java/org/checkerframework/framework/util/DefaultAnnotationFormatter.java
@@ -86,12 +86,14 @@ public class DefaultAnnotationFormatter implements AnnotationFormatter {
                 boolean notfirst = false;
                 for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> arg :
                         args.entrySet()) {
-                    if (notfirst) {
-                        sb.append(", ");
+                    if (!"{}".equals(arg.getValue().toString())) {
+                        if (notfirst) {
+                            sb.append(", ");
+                        }
+                        notfirst = true;
+                        sb.append(arg.getKey().getSimpleName() + "=");
+                        formatAnnotationMirrorArg(arg.getValue(), sb);
                     }
-                    notfirst = true;
-                    sb.append(arg.getKey().getSimpleName() + "=");
-                    formatAnnotationMirrorArg(arg.getValue(), sb);
                 }
             }
             sb.append(")");

